src/server/scripts/EasternKingdoms/ScarletMonastery/boss_mograine_and_whitemane.cpp

```diff
@@ -61,321 +53,391 @@ enum Spells
     SPELL_POWERWORDSHIELD        = 22187
 };
 
-class boss_scarlet_commander_mograine : public CreatureScript
+enum Events
 {
-public:
-    boss_scarlet_commander_mograine() : CreatureScript("boss_scarlet_commander_mograine") { }
-
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return GetScarletMonasteryAI<boss_scarlet_commander_mograineAI>(creature);
-    }
+    /// Both
+    EVENT_RESET_KILL_TALK = 1,
 
-    struct boss_scarlet_commander_mograineAI : public ScriptedAI
-    {
-        boss_scarlet_commander_mograineAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            instance = creature->GetInstanceScript();
-        }
+    /// Mograine
+    EVENT_CRUSADER_STRIKE = 2,
+    EVENT_HAMMER_OF_JUSTICE,
 
-        void Initialize()
-        {
-            CrusaderStrike_Timer = 10000;
-            HammerOfJustice_Timer = 10000;
-            _bHasDied = false;
-            _bHeal = false;
-            _bFakeDeath = false;
-        }
+    /// Whitemane
+    EVENT_HEAL = 2,
+    EVENT_POWER_WORD_SHIELD,
+    EVENT_HOLY_SMITE,
+};
 
-        InstanceScript* instance;
+enum Points
+{
+    POINT_WHITEMANE_MOVE_TO_MOGRAINE = 1,
+};
 
-        uint32 CrusaderStrike_Timer;
-        uint32 HammerOfJustice_Timer;
+const Position WhitemaneIntroMovePos(1163.113370f, 1398.856812f, 32.527786f);
 
-        bool _bHasDied;
-        bool _bHeal;
-        bool _bFakeDeath;
+/// Scarlet Commander Mograine - 3976
+struct boss_scarlet_commander_mograine : public ScriptedAI
+{
+    public:
+        boss_scarlet_commander_mograine(Creature* creature) : ScriptedAI(creature)
+        {
+            m_CanTalkKill = true;
+            m_CanDie = false;
+            m_FakeDeath = false;
+        }
 
         void Reset() override
         {
-            Initialize();
+            m_CanTalkKill = true;
```

Comment:
keeping the old Initialize() way reduce code duplication, so these variables are set to their default value only in a function